crates/sui-core/src/consensus_adapter.rs

```diff
@@ -899,6 +908,18 @@ impl ConsensusAdapter {
                         processed.expect("Storage error when waiting for transaction executed in checkpoint");
                         self.metrics.sequencing_certificate_processed.with_label_values(&["checkpoint"]).inc();
                     }
+                    processed = async {
+                        // If the transaction is a checkpoint signature, we can also wait to get notified when a checkpoint with equal or higher sequence
+                        // number has been already synced. This way we don't try to unnecessarily sequence the signature for an already verified checkpoint.
+                        if let Some(checkpoint_sequence_number) = checkpoint_sequence_number {
+                            epoch_store.synced_checkpoint_notify(checkpoint_sequence_number).await
+                        } else {
+                            Ok(())
+                        }
+                    }, if checkpoint_sequence_number.is_some() => {
```

Comment:
It seems this can be simplified, for example why not `processed = epoch_store.synced_checkpoint_notify(checkpoint_sequence_number.unwrap()), if checkpoint_sequence_number.is_some()`? But Mark's suggestion could be better.

Reply:
@mwtian actually that will make the code fail (that's how I did it initially as well), since the `checkpoint_sequence_number.unwrap()` always gets evaluated even when the branch is disabled. Apparently Rust works like that. That's why I had to do the Some/None approach.